src/Lean/Server/InfoUtils.lean

```diff
@@ -95,57 +118,47 @@ def InfoTree.smallestInfo? (p : Info → Bool) (t : InfoTree) : Option (ContextI
 
 /-- Find an info node, if any, which should be shown on hover/cursor at position `hoverPos`. -/
 partial def InfoTree.hoverableInfoAt? (t : InfoTree) (hoverPos : String.Pos) : Option (ContextInfo × Info) :=
-  t.smallestInfo? fun i =>
-    if let (some pos, some tailPos) := (i.pos?, i.tailPos?) then
-      if pos ≤ hoverPos ∧ hoverPos < tailPos then
-        match i with
-        | Info.ofTermInfo ti =>
-          !ti.expr.isSyntheticSorry &&
```

Comment:
I believe a long time ago the elaborator would produce synthetic sorrys with valid positions in certain places (maybe structure initializer fields?) and those would produce nonsensical hovers. A lot has changed since then regarding Syntax positions, so maybe not a problem anymore, not sure. Btw, I will look over this PR later.

Reply:
@Kha why isn't the filter needed anymore? Does it not show various crazy internal terms anymore because of the changes made to synthetic positions?

Reply:
Yes, it does look a bit funny when hovering over e.g. quotations. I've now refined the formatting logic to output the term only if it is an atomic format object (but always output the type), please tell me what you think.

Reply:
Also added a test for the synthetic sorry.

Reply:
Awesome, thanks! (and we do still have the synthetic sorries in the tree).